src/components/application_manager/src/hmi_capabilities_impl.cc

```diff
@@ -1037,10 +1036,29 @@ struct JsonCapabilitiesGetter {
   }
 
  private:
-  Json::Value& json_default_node_;
-  Json::Value& json_cache_node_;
+  Json::Value json_default_node_;
+  Json::Value json_cache_node_;
 };
 
+/**
+ * @brief Gets the JSON node related to the specified interface name.
+ * @param interface_name Interface name which JSON node should be retreive from
+ * the main JSON node
+ * @return JsonCapabilitiesGetter instance initialized by default JSON node and
+ * cache JSON node for the specified interface
+ */
+
+JsonCapabilitiesGetter GetInterfaceGetter(
+    const char* interface_name, JsonCapabilitiesGetter& json_root_getter) {
+  auto interface_default_node =
+      json_root_getter.GetMainJsonMember(interface_name);
+  auto interface_cache_node =
+      json_root_getter.GetCachedJsonMember(interface_name);
+
+  return std::move(
```

Comment:
I think this is not a good use of std::move, this function is assigning to new values and compiler return value optimizations would be faster.

http://thbecker.net/articles/rvalue_references/section_06.html